build/source/engine/allocspace.f90

```diff
@@ -425,20 +448,20 @@ subroutine allocateDat_int(metadata,nSnow,nSoil,nLayers, & ! input
   ! allocate structures
   else
    select case(metadata(iVar)%vartype)
-    case(iLookVarType%scalarv); allocate(varData%var(iVar)%dat(1),stat=err)
-    case(iLookVarType%wLength); allocate(varData%var(iVar)%dat(nBand),stat=err)
-    case(iLookVarType%midSnow); allocate(varData%var(iVar)%dat(nSnow),stat=err)
-    case(iLookVarType%midSoil); allocate(varData%var(iVar)%dat(nSoil),stat=err)
-    case(iLookVarType%midToto); allocate(varData%var(iVar)%dat(nLayers),stat=err)
-    case(iLookVarType%ifcSnow); allocate(varData%var(iVar)%dat(0:nSnow),stat=err)
-    case(iLookVarType%ifcSoil); allocate(varData%var(iVar)%dat(0:nSoil),stat=err)
-    case(iLookVarType%ifcToto); allocate(varData%var(iVar)%dat(0:nLayers),stat=err)
-    case(iLookVarType%routing); allocate(varData%var(iVar)%dat(nTimeDelay),stat=err)
-    case(iLookVarType%unknown); allocate(varData%var(iVar)%dat(0),stat=err)  ! unknown=special (and valid) case that is allocated later (initialize with zero-length vector)
+    case(iLookVarType%scalarv); allocate(varData%var(iVar)%dat(1),stat=err,errmsg=cmessage)
```

Comment:
Nice. I did not realize that there was an `errmsg` argument to the allocate routine.